### PR TITLE
NEP: Updated NEP-35 with keyword-only instruction

### PR DESCRIPTION
**Pull request by pentschev**
During the implementation phase in https://github.com/numpy/numpy/pull/16935#discussion_r459668002, it was suggested to make `like=` as a keyword-only, updating the NEP accordingly.

### CODE REVIEW

**Comment by pentschev** [comment]
cc @shoyer @hameerabbasi @eric-wieser @rgommers @seberg

**Review by shoyer — APPROVED** [review]
Looks great, thanks for updating!

**Review by hameerabbasi — APPROVED** [review]
Thanks a lot!

**Comment by hameerabbasi** [comment]
Was this closed by accident?

**Comment by pentschev** [comment]
Very much by accident, I hadn't even seen that, thanks for pinging @hameerabbasi !

**Comment by pentschev** [comment]
It seems like the s390x build is failing and there's no output, not sure who I should ping about such CI issues or if they can just be ignored.

**Comment by rgommers** [comment]
@pentschev I restarted it, but in principle you can ignore it. It often happens that CI fails on PRs for reasons unrelated to that PR, if that happens we're fine to continue work or merge a PR, and deal with the CI failure elsewhere.

**Comment by pentschev** [comment]
Thanks @rgommers , it does indeed seem like there's a problem with s390x, it's stuck there in the [current build](https://travis-ci.org/github/numpy/numpy/builds/714969388), and also stuck there in the [build from #16935](https://travis-ci.org/github/numpy/numpy/builds/715089051) which has been running for several hours now.

With that said, if there are no further comments/requests, this PR is ready to merge from my side.

File: doc/neps/nep-0035-array-creation-dispatch-with-array-function.rst [hunk 1]
```diff
@@ -121,7 +125,8 @@ the module too.
         module='numpy', docs_from_dispatcher=False, verify=False)
 
     @array_function_nodocs_from_c_func_and_dispatcher(_multiarray_umath.array)
-    def array(a, dtype=None, copy=True, order='K', subok=False, ndmin=0, like=None):
+    def array(a, dtype=None, copy=True, order='K', subok=False, ndmin=0, *,
```
Comment (BvB93) [inline doc/neps/nep-0035-array-creation-dispatch-with-array-function.rst:128]:
```suggestion
    def array(a, dtype=None, *, copy=True, order='K', subok=False, ndmin=0,
```
Minor nitpick: all arguments after `dtype` are keyword-only; not just `like`. https://github.com/numpy/numpy/blob/6bed9a91c1be207d2b69f13355449347d22c850b/numpy/core/_add_newdocs.py#L789

Comment (pentschev) [reply doc/neps/nep-0035-array-creation-dispatch-with-array-function.rst]: Adjusted this now.

**Comment by pentschev** [comment]
Any last comments here? If not, this should be ready to go.

File: doc/neps/nep-0035-array-creation-dispatch-with-array-function.rst [hunk 2]
```diff
@@ -93,12 +97,12 @@ with ``overrides.array_function_dispatch``:
...
 .. code:: python
 
-    def _asarray_decorator(a, dtype=None, order=None, like=None):
+    def _asarray_decorator(a, dtype=None, order=None, *, like=None):
         return (like,)
 
     @set_module('numpy')
     @array_function_dispatch(_asarray_decorator)
-    def asarray(a, dtype=None, order=None, like=None):
+    def asarray(a, dtype=None, order=None, *, like=None):
```
Comment (shoyer) [inline doc/neps/nep-0035-array-creation-dispatch-with-array-function.rst:105]: There is a case for making `order` keyword only as well, but this is certainly sufficiently for now.

**Comment by shoyer** [comment]
OK, I will merge this as soon as CI passes verifying that building the doc page works!

File: doc/neps/nep-0035-array-creation-dispatch-with-array-function.rst (hunk 2 above)
Comment (pentschev) [reply doc/neps/nep-0035-array-creation-dispatch-with-array-function.rst:105]: IMO, while it's good to have things match for future reference, they're of lesser importance given this is just an example, I suspect these will anyway drift from future API changes.

**Comment by shoyer** [comment]
thanks @pentschev !

**Comment by pentschev** [comment]
Thanks everyone for reviews and @shoyer for merging.

**Pull request merged**